### ISSUE TITLE
typecheck fails when used with esbuild 0.15.3

### ISSUE THREAD

**Issue by wydengyre** [lucacasonato/esbuild_deno_loader#21]
```
error: TS2769 [ERROR]: No overload matches this call.
  The last overload gave the following error.
    Type 'import("https://deno.land/x/esbuild@v0.14.51/mod.d.ts").Plugin' is not assignable to type 'import("https://deno.land/x/esbuild@v0.15.3/mod.d.ts").Plugin'.
      Types of property 'setup' are incompatible.
        Type '(build: import("https://deno.land/x/esbuild@v0.14.51/mod.d.ts").PluginBuild) => void | Promise<void>' is not assignable to type '(build: import("https://deno.land/x/esbuild@v0.15.3/mod.d.ts").PluginBuild) => void | Promise<void>'.
          Types of parameters 'build' and 'build' are incompatible.
            Type 'import("https://deno.land/x/esbuild@v0.15.3/mod.d.ts").PluginBuild' is not assignable to type 'import("https://deno.land/x/esbuild@v0.14.51/mod.d.ts").PluginBuild'.
              The types of 'initialOptions.stdin' are incompatible between these types.
                Type 'import("https://deno.land/x/esbuild@v0.15.3/mod.d.ts").StdinOptions | undefined' is not assignable to type 'import("https://deno.land/x/esbuild@v0.14.51/mod.d.ts").StdinOptions | undefined'.
                  Type 'import("https://deno.land/x/esbuild@v0.15.3/mod.d.ts").StdinOptions' is not assignable to type 'import("https://deno.land/x/esbuild@v0.14.51/mod.d.ts").StdinOptions'.
                    Types of property 'contents' are incompatible.
                      Type 'string | Uint8Array' is not assignable to type 'string'.
                        Type 'Uint8Array' is not assignable to type 'string'.
    plugins: [denoPlugin({ importMapURL })],

```

**Comment by jsejcksn** [comment]
Similarly with the latest version of esbuild `0.15.7`:

```
No overload matches this call.
  The last overload gave the following error.
    Type 'import("https://deno.land/x/esbuild@v0.14.51/mod").Plugin[]' is not assignable to type 'import("https://deno.land/x/esbuild@v0.15.7/wasm").Plugin[]'.
      Type 'import("https://deno.land/x/esbuild@v0.14.51/mod").Plugin' is not assignable to type 'import("https://deno.land/x/esbuild@v0.15.7/wasm").Plugin'.
        Types of property 'setup' are incompatible.
          Type '(build: import("https://deno.land/x/esbuild@v0.14.51/mod").PluginBuild) => void | Promise<void>' is not assignable to type '(build: import("https://deno.land/x/esbuild@v0.15.7/wasm").PluginBuild) => void | Promise<void>'.
            Types of parameters 'build' and 'build' are incompatible.
              Type 'import("https://deno.land/x/esbuild@v0.15.7/wasm").PluginBuild' is not assignable to type 'import("https://deno.land/x/esbuild@v0.14.51/mod").PluginBuild'.
                The types of 'initialOptions.stdin' are incompatible between these types.
                  Type 'import("https://deno.land/x/esbuild@v0.15.7/wasm").StdinOptions | undefined' is not assignable to type 'import("https://deno.land/x/esbuild@v0.14.51/mod").StdinOptions | undefined'.
                    Type 'import("https://deno.land/x/esbuild@v0.15.7/wasm").StdinOptions' is not assignable to type 'import("https://deno.land/x/esbuild@v0.14.51/mod").StdinOptions'.
                      Types of property 'contents' are incompatible.
                        Type 'string | Uint8Array' is not assignable to type 'string'.
                          Type 'Uint8Array' is not assignable to type 'string'.deno-ts(2769)
```

**Comment by jsejcksn** [comment]
I'm not able to test at the moment, but this might have been resolved by the dependency updates in #23.
